File: app/src/main/java/com/example/xiangxueketang/lesson2/MainActivity.java
package com.example.xiangxueketang.lesson2;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.widget.TextView;

import com.example.xiangxueketang.R;
import com.example.xiangxueketang.lesson2.annotation.InjectView;
import com.example.xiangxueketang.lesson2.utils.InjectViewUtils;

public class MainActivity extends AppCompatActivity {
    //模仿BufferKnife获取控件id
    @InjectView(R.id.text_view)
     TextView mTextView;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        InjectViewUtils.injectView(this);
        mTextView.setText("模仿BufferKnife获取控件id");
    }
}

File: kotlintest/src/main/java/com/example/kotlintest/MainActivity.kt
package com.example.kotlintest

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import com.example.kotlintest.singleton.StaticInnerSingleton

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        StaticInnerSingleton.instance.show()
    }
}